realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -316,6 +318,11 @@ private static void initializeRealm(Context context, String userAgent) {
             }
             checkFilesDirAvailable(context);
 
+            // https://github.com/realm/realm-java/issues/7640
+            if ((Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) && context.getPackageManager().isInstantApp()) {
```

Comment:
What does this check do? `(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)`

Reply:
It looks like https://developer.android.com/reference/android/content/pm/PackageManager#isInstantApp() was added in API 26 and we support back to 16. We should probably delegate this to a check that only runs if API is above 25. If there is no way to check below 26, that is just unfortunate, but nothing we can do I guess.

Reply:
Doh...was reading the O as 0 🙈 ... Nevermind me

Reply:
I found that there is a `PackageManagerCompat` that has a retrocompatible `isInstantApp` method. The issue is that is part of the optional GMS play services library. I am calling this library via reflection if available, if not fallbacking into the system package manager.